app/src/main/java/rightmesh/left/io/gpstracker/RightMeshConnector.java

```diff
@@ -0,0 +1,203 @@
+                    }
+                });
+                androidMeshManager.on(PEER_CHANGED, event -> {
+                    if (peerchangedListener != null) {
+                        peerchangedListener.onPeerChange(event);
+                    }
+                });
+            } catch (RightMeshException.RightMeshServiceDisconnectedException sde) {
+                Log.e(TAG, "Service disconnected while binding, with message: "
+                        + sde.getMessage());
+            } catch (RightMeshException rme) {
+                Log.e(TAG, "MeshPort already bound, with message: " + rme.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Set listener for data receive event.
+     *
+     * @param listener a callback
+     */
+    public void setOnDataReceiveListener(OnDataReceiveListener listener) {
+        this.dataReceiveListener = listener;
+    }
+
+    /**
+     * Set listener for peer change event.
+     *
+     * @param listener a callback
+     */
+    public void setOnPeerChangedListener(OnPeerChangedListener listener) {
+        this.peerchangedListener = listener;
+    }
+
+    /**
+     * Set listener for my MeshId receiving event.
+     *
+     * @param listener a callback
+     */
+    public void setOnConnectSuccessListener(OnConnectSuccessListener listener) {
+        this.connectSuccessListener = listener;
+    }
+
+    /**
+     * Navigate to Rightmesh Wallet app.
+     *
+     * @throws RightMeshException can't navigate
+     */
+    public void toRightMeshWalletActivty() throws RightMeshException {
+        this.androidMeshManager.showSettingsActivity();
+    }
+
+    /**
+     * Send data to target device.
+     *
+     * @param targetMeshId Target meshId.
+     * @param payload      data need to send.
+     * @return Data Id
+     * @throws RightMeshException.RightMeshServiceDisconnectedException Service disconnected.
+     * @throws RightMeshException                                       Can't find next hop.
+     */
+    public int sentDataReliable(MeshId targetMeshId, byte[] payload) throws RightMeshException,
+            RightMeshException.RightMeshServiceDisconnectedException {
+        return androidMeshManager.sendDataReliable(androidMeshManager.getNextHopPeer(targetMeshId),
+                meshPort, payload);
+    }
+
+    /**
+     * Close RightMesh connection when activity is destroyed.
+     */
+    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+    public void stop() {
+        try {
+            androidMeshManager.stop();
+        } catch (RightMeshException.RightMeshServiceDisconnectedException e) {
+            Logger.fatal(TAG, "Service disconnected before stopping AndroidMeshManager "
+                    + "with message" + e.getMessage());
+        }
+    }
+
+    /**
+     * Resume RightMesh connection on activity resume.
+     */
+    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
+    public void resumse() {
```

Comment:
resume
```suggestion
    public void resume() {
```

Reply:
I fixed all typo errors in the next version